### PR TITLE
hp2p: new package

### PR DESCRIPTION
**Pull request by rfbgo**

### CODE REVIEW

**Comment by rfbgo** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/hp2p/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://github.com/cea-hpc/hp2p/releases/download/4.1/hp2p-4.1.tar.gz"
+    git = "https://github.com/cea-hpc/hp2p.git"
+
+    version("4.1", sha256="e74fa1d442f4378a31f4b875760aaad98e23f6942f7de4cc1702ed9e95585c5e")
+
+    depends_on("mpi", type=("build", "link", "run"))
+
+    def configure_args(self):
+        spec = self.spec
+        args = ["CC={0}".format(spec["mpi"].mpicc), "CXX={0}".format(spec["mpi"].mpicxx)]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hp2p/package.py]: We now encourage use of Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) where appropriate.

File: var/spack/repos/builtin/packages/hp2p/package.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class Hp2p(AutotoolsPackage):
+    """HP2P (Heavy Peer To Peer) benchmark is a test which performs MPI
+    Point-to-Point non-blocking communications between all MPI processes. Its
+    goal is to measure the bandwidths and the latencies in a situation where
+    the network is busy. This benchmark can help to detect network problems
+    like congestions or problems with switches or links.
+    """
+
+    home = "https://github.com/cea-hpc/hp2p"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hp2p/package.py]: This needs to be the `homepage` property:

```suggestion
    homepage = "https://github.com/cea-hpc/hp2p"
```

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/hp2p/package.py]: Confirmed the  homepage exists (when specified using `homepage`).

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the version sha256. 

Please correct the home page property name. Switching to using f-strings for the args is encouraged but not required. You are also encouraged to add yourself as the maintainer for the package (https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers).

File: var/spack/repos/builtin/packages/hp2p/package.py (hunk 1 above)
Comment (rfbgo) [reply var/spack/repos/builtin/packages/hp2p/package.py]: Thanks for the feedback! Agreed and applied, ptal :)

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**